pandas/tests/frame/test_indexing.py

```diff
@@ -2936,6 +2936,20 @@ def test_where_callable(self):
         tm.assert_frame_equal(result,
                               (df + 2).where((df + 2) > 8, (df + 2) + 10))
 
+    def test_where_tz_values(self):
+        dts1 = date_range('20150101', '20150105', tz='America/New_York')
+        df1 = DataFrame({'date': dts1})
+        dts2 = date_range('20150103', '20150107', tz='America/New_York')
+        df2 = DataFrame({'date': dts2})
+        mask = DataFrame(True, index=df1.index, columns=df2.columns)
+        mask.iloc[3:] = False
+        result = df1.where(mask, df2)
+        dts3 = DatetimeIndex(['20150101', '20150102', '20150103',
```

Comment:
ideally just contruct what you can w/o using temporaries.

ideally call this expected